docs/knowledge_base/tips/using_tables/index.md

```diff
@@ -0,0 +1,327 @@
+some columns to have a default or set value.
+
+## Deleting (on_delete)
+
+Finally, the deletion process works as follows:
+
+![Deleting](tables-on_delete.gif){width=100%}
+
+We can implement basic functionality with the following code:
+
+```py
+def food_df_on_delete(state, var_name, action, payload):
+    index = payload["index"] # row index
+ 
+    state.food_df = state.food_df.drop(index=index)
+    notify(state, "E", f"Deleted row at index '{index}'")
+ 
+main_md = Markdown("<|{food_df}|table|on_delete=food_df_on_delete|>")
+```
+
+## The Complete App
+
+Once more, please note that this tip article doesn't cover all the features of tables 
+in Taipy comprehensively. If you're seeking a feature that we didn't address here, 
+be sure to refer to the [documentation](../../../manuals/gui/viselements/table/) for more information!
+
+Lastly, here's the code that combines all the features we discussed in this article, 
+used to create the app shown at the beginning of the article:
+
+```py
+from taipy.gui import Gui, Markdown, notify
+import pandas as pd
+ 
+ 
+food_df = pd.DataFrame({
+    "Meal": ["Lunch", "Dinner", "Lunch", "Lunch", "Breakfast", "Breakfast", "Lunch", "Dinner"],
+    "Category": ["Food", "Food", "Drink", "Food", "Food", "Drink", "Dessert", "Dessert"],
+    "Name": ["Burger", "Pizza", "Soda", "Salad", "Pasta", "Water", "Ice Cream", "Cake"],
+    "Calories": [300, 400, 150, 200, 500, 0, 400, 500],
+})
+ 
+ 
+def food_df_on_edit(state, var_name, action, payload):
+    index = payload["index"] # row index
+    col = payload["col"] # column name
+    value = payload["value"] # new value cast to the column type
+    user_value = payload["user_value"] # new value as entered by the user
+ 
+    old_value = state.food_df.loc[index, col]
+    new_food_df = state.food_df.copy()
+    new_food_df.loc[index, col] = value
+    state.food_df = new_food_df
+    notify(state, "I", f"Edited value from '{old_value}' to '{value}'. (index '{index}', column '{col}')")
+ 
+ 
+def food_df_on_delete(state, var_name, action, payload):
+    index = payload["index"] # row index
+ 
+    state.food_df = state.food_df.drop(index=index)
+    notify(state, "E", f"Deleted row at index '{index}'")
+ 
+ 
+def food_df_on_add(state, var_name, action, payload):
+    empty_row = pd.DataFrame([[None for _ in state.food_df.columns]], columns=state.food_df.columns)
+    state.food_df = pd.concat([empty_row, state.food_df], axis=0, ignore_index=True)
+ 
+    notify(state, "S", f"Added a new row.")
+ 
+ 
+table_properties = {
+    "class_name": "rows-bordered",
+    "filter": True,
+    "on_edit": food_df_on_edit,
+    "on_delete": food_df_on_delete,
+    "on_add": food_df_on_add,
+    "group_by[Category]": True,
+    "apply[Calories]": "sum",
+}
+ 
+ 
+main_md = Markdown("""
+# Daily Calorie Tracker
+ 
+<|{food_df}|table|properties=table_properties|>
+""")
+ 
+Gui(page=main_md).run(dark_mode=False)
```

Comment:
I would change this to use the tp.run(Gui(page=main_md, dark_mode=False)) syntax. @FabienLelaquais @FlorianJacta what do you think?

Reply:
There's no need for that. That tips can be used as is with only taipy-gui.
(but please remove the "dark_mode=False" part)